crates/erg_parser/parse.rs

```diff
@@ -246,17 +263,89 @@ impl Parser {
         }
     }
 
-    fn skip_and_throw_syntax_err(&mut self, caused_by: &str) -> ParseError {
+    fn until_dedent(&mut self) {
+        while let Some(t) = self.peek() {
+            match t.kind {
+                Dedent => {
+                    self.skip();
+                    return;
+                }
```

Comment:
In the case of multiple nesting, this may not be sufficient. It must be skipped until all Dedents are removed.